tutorials/making_plugins.rst

```diff
@@ -0,0 +1,247 @@
+other will be a custom dock added to the editor.
+
+Creating a plugin
+~~~~~~~~~~~~~~~~~
+
+Before starting, create a new empty project wherever you want. This will serve
+as base to develop and test the plugins.
+
+The first thing you need to do is to create a new plugin that the editor can
+understand as such. For that you need to files: ``plugin.cfg`` for the
+configuration and a custom GDScript with the functionality.
+
+The plugins have a standard path of ``addons/plugin_name`` inside the project
+folder. So create the folder ``my_custom_dock`` inside ``addons``. So you'll
+have a directory structure like this:
+
+.. image:: /img/making_plugins-my_custom_mode_folder.png
+
+To make the ``plugin.cfg`` file, open your favorite text editor with a blank
+file. Godot is not able (yet) to open text files besides scripts, so this must
+be done in an external editor. Add the following structure to you plugin::
+
+    [plugin]
+
+    name="My Custom Node"
+    description="A custom node made to extend the Godot Engine."
+    author="Your Name Here"
+    version="1.0"
+    script="custom_node.gd"
+
+This is a simple ``ini`` file with metadata about your plugin. You need to set
+up the name and description so users can understand what it does. Add your
+own name so you can be properly credited. A version number so users can see if
+they have an outdated version (if you are unsure on how to make the version
+number, check `SemVer <http://semver.org/>`_). And finally a main script file
+to load when your plugin is active.
+
+The script file
+^^^^^^^^^^^^^^^
+
+Open the script editor (F3) and create a new GDScript file called
+``custom_dock.gd`` inside the ``my_custom_node`` folder. This script is special
+and it has two requirements: it must be a ``tool`` script and it has to
+inherit from :ref:`class_EditorPlugin`.
+
+It's important to deal with initialization and clean-up of resources. So a good
+practice is to use the virtual function
+:ref:`_enter_tree() <class_Node__enter_tree>` to initialize your plugin and
+:ref:`_exit_tree() <class_Node__exit_tree>` to clean it up. You can delete the
+default GDScript template from your file and replace it with the following
+structure:
+
+.. _doc_making_plugins_template_code:
+.. code-block:: python
+
+    tool
+    extends EditorPlugin
+
+    func _enter_tree():
+        # Initialization of the plugin goes here
+        pass
+
+    func _exit_tree():
+        # Clean-up of the plugin goes here
+        pass
+
+This is a good template to use when devising new plugins.
+
+A custom node
+~~~~~~~~~~~~~~~~~~~~
+
+Sometimes you want a certain behavior in many nodes. Maybe a custom scene
+or control that can be reused. Instancing is helpful in a lot of case but
+sometimes it can be cumbersome, especially if you're using it between many
+projects. A good solution to this is to make a plugin that adds a node with a
+custom behavior.
+
+To create a new node type, you can avail of the can avail of the funcion
```

Comment:
I like how you mentioned the typo but didn't say I wrote "can avail of the" twice :laughing: